packages/orchestration/src/exos/portfolio-holder-kit.js

```diff
@@ -0,0 +1,200 @@
+import { M, mustMatch } from '@endo/patterns';
+import { E } from '@endo/far';
+import { VowShape } from '@agoric/vow';
+import { Fail } from '@endo/errors';
+import {
+  TopicsRecordShape,
+  PublicTopicShape,
+} from '@agoric/zoe/src/contractSupport/topics.js';
+import { makeScalarBigMapStore } from '@agoric/vat-data';
+
+const { fromEntries } = Object;
+
+/**
+ * @import {MapStore} from '@agoric/store';
+ * @import {VowTools} from '@agoric/vow';
+ * @import {ResolvedPublicTopic, ResolvedTopicsRecord} from '@agoric/zoe/src/contractSupport/topics.js';
+ * @import {Zone} from '@agoric/zone';
+ * @import {CopyMap} from '@endo/patterns';
+ * @import {OrchestrationAccount} from '@agoric/orchestration';
+ * @import {ResolvedContinuingOfferResult} from '../utils/zoe-tools.js';
+ */
+
+/**
+ * @typedef {{
+ *   accounts: MapStore<string, OrchestrationAccount<any>>;
+ *   publicTopics: MapStore<string, ResolvedPublicTopic<unknown>>;
+ * }} PortfolioHolderState
+ */
+
+const ChainNameM = M.string();
+
+const AccountEntriesShape = M.or(
+  M.arrayOf([M.string(), M.remotable('OrchestrationAccount')]),
+  M.mapOf(M.string(), M.remotable('OrchestrationAccount')),
+);
+const PublicTopicsShape = M.or(
+  M.arrayOf([M.string(), PublicTopicShape]),
+  M.mapOf(M.string(), PublicTopicShape),
+);
```

Comment:
`FooShape` should match a `Foo` type. I see you need the maker function to accept entries or a dict. Please pick one. I suggest the entries format. I don't think the `M.mapOf` works the way you're expecting and that codepath has no coverage in this branch.

Reply:
> If you're suggesting `PublicTopicEntriesShape`, agree and will make the edit.

👍 

> There is coverage for `.mapOf` here:

Missed that, thanks

> It felt weird narrowing the allowed parameters to `.addAll`.

Not weird. The caller of this doesn't know that `addAll` is happening.


> If we are to leave both a map or array of entries as valid parameters, would you like me to rename the shape? `AccountMapEntriesShape`, something else?

I would prefer to pick one, but won't block on it. If you do keep it wide, yes the shape should convey the or-ness.